### ISSUE TITLE
Reimported modules are detected as missing by `import/names`

### ISSUE THREAD

**Issue by maknahar** [benmosher/eslint-plugin-import#1584]
Given:

```js
// ./foo/foo1.js
export const foo1 = "I'm so foo1"
```

```js
// ./foo/foo2.js
export const foo2 = "I'm so foo2"
```

```js
// ./foo/index.js
import foo1 from './foo1'
import foo2 from './foo2'

module.exports = {
  foo1, foo2
}
```

The following is considered invalid:

```js
// ./bar.js
import { foo1, foo2 } from './foo'
```

Error: 

>ESLint: foo1 not found in './foo'(import/named)
>ESLint: foo2 not found in './foo'(import/named)

However, the following is considered valid

```js
// ./bar.js
import * as bar from './foo'
```

is it done intentionally? And if so, do we have an option to disable errors? 

My configuration:
```
{
    "env": {
        "browser": true,
        "jest": true
    },
    "parserOptions": {
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "extends": [
        "airbnb",
        "plugin:prettier/recommended"
    ],
    "plugins": [
        "react"
    ],
    "rules": {
        "class-methods-use-this": 0,
        "import/no-named-as-default": 0,
        "react/jsx-filename-extension": [
            "error",
            {
                "extensions": [
                    ".js",
                    ".jsx"
                ]
            }
        ]
    }
}
```

Eslint version: v6.8.0

**Comment by ljharb** [comment]
`import * as bar` works even if there's zero exports, so that being valid isn't relevant.

I think you're running into some form of what #1226 attempts to fix.
